basho/riak-java-client
# Fix for #572. Refresh connection IP address info with (possibly) new DNS info when creating a new connection.

## 1. Problem

If you used the Java Client in a cloud environment where IP addresses for DNS entries could change often, the Java client would not expire the old DNS lookup to get the newest IP address.
## 2. Solution

Force a refresh any time a connection is made outside of the cluster/node `start()` method. (Lookup is done once at start). 
https://github.com/basho/riak-java-client/pull/573/files#diff-28c20364743c5fc78c7c41c7c8b89e88R293

Also added a log warning if they have their DNS cache setting set to "infinity", which would also cause this issue: https://github.com/basho/riak-java-client/pull/573/files#diff-28c20364743c5fc78c7c41c7c8b89e88R925
## 3. Test
1. Setup a "good" mock Netty `Bootstrap` / `ChannelFuture` / `Channel` with a `RiakNode`
2. Start RiakNode, should do address lookup #1. 
3. Fetch/return 1 good connection. 
4. Force the mock to return `false` for `channel.isOpen()` when we get the next connection, which should force a second lookup.
5. Get next connection.
6. Verify that there were two address lookups.

https://github.com/basho/riak-java-client/pull/573/files#diff-4361a4b26a0905c4782f996ce653ccaaR416
